### PR TITLE
Fix compaction with cluster snapshot deletion

### PR DESCRIPTION
**Pull request by zhanglingzhe0820**
## Behavior
Compaction run with cluster snapshot together and meet error.

## Reason
If run as below:
(1) Start compaction task;
(2) Cluster pull snapshot;
(3) The snapshot pulled is fully overlapped with some merging local file and delete them.
(4) The compaction cannot find the file to be read and exit.
The temp file will not be deleted.

## Solution
1、Add restore logic when meet exception.

### CODE REVIEW

**Review by neuyilan — COMMENTED** [review]
LGTM

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/utils/CompactionUtils.java [hunk 1]
```diff
@@ -271,6 +271,9 @@ public static void merge(
       for (TsFileResource tsFileResource : tsFileResources) {
         TsFileSequenceReader reader =
             buildReaderFromTsFileResource(tsFileResource, tsFileSequenceReaderMap, storageGroup);
+        if (reader == null) {
+          throw new IOException();
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/utils/CompactionUtils.java:275]: Add some message in this exception.

**Review by jt2594838 — APPROVED** [review]

**Comment by jt2594838** [comment]
There is a failed test, please look into it.

**Comment by zhanglingzhe0820** [comment]
> There is a failed test, please look into it.

This test has nothing to do with this pr.

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/utils/CompactionUtils.java (hunk 1 above)
Comment (zhanglingzhe0820) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/utils/CompactionUtils.java:275]: okay

**Pull request merged**